tests/PaymentBundle/Controller/PaymentControllerTest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\PaymentBundle\Tests\Controller;
+
+use PHPUnit\Framework\TestCase;
+use Sonata\Component\Basket\BasketFactoryInterface;
+use Sonata\PaymentBundle\Controller\PaymentController;
+
+final class PaymentControllerTest extends TestCase
+{
+    public function testInstanceBasketInterfacePaymentAction(): void
+    {
+        $class = new \ReflectionClass(PaymentController::class);
+        $method = $class->getMethod('getBasketFactory');
+        $method->setAccessible(true);
+
+        $basketFactoryInterface = $this->createMock(BasketFactoryInterface::class);
+
+        $paymentController = $this->createPaymentController($basketFactoryInterface);
+
+        $this->assertSame($basketFactoryInterface, $method->invoke($paymentController));
+    }
+
+    public function createPaymentController($basketFactory = null)
+    {
+        if (null === $basketFactory) {
+            $basketFactory = $this->createMock(BasketFactoryInterface::class);
+        }
+
+        return new PaymentController($basketFactory);
```

Comment:
The tests are failing because you need to pass all dependencies or mark this test as legacy (with `@group legacy`)